packages/hosted-form-v2/src/create-hosted-form-service.ts

```diff
@@ -8,6 +10,9 @@ import HostedFormService from './hosted-form-service';
  * @param host - Host url string parameter.
  * @returns An instance of `HostedFormService`.
  */
-export function createHostedFormService(host: string) {
-    return new HostedFormService(host, new HostedFormFactory());
+export function createHostedFormService(
+    host: string,
+    paymentIntegrationService: PaymentIntegrationService,
+) {
+    return new HostedFormService(host, new HostedFormFactory(paymentIntegrationService));
```

Comment:
Can `paymentIntegrationService` be optional? 
This function is used in manual-order-ui ([ref](https://github.com/bigcommerce/manual-order-ui/blob/main/src/Payment/components/PaymentMethod/CreditCardPaymentMethod/CreditCardIntegrationService.ts#L24)) from where we do not need to pass `paymentIntegrationService` and this will fail in that case.

Reply:
Hi @richapatel1510 
I think not. As `createHostedFormService` function returns HostedForm which requires paymentIntegrationService and HostedFormOrderDataTransformer also requires it.
Can we just add `paymentIntegrationService` to `createHostedFormService` use case which you linked?

Reply:
Hi @bc-sebastianszafraniec ,
Can this be initialised in the `HostedFormFactory`'s constructor itself, without the need to pass it from here?

Reply:
Hi @richapatel1510 I removed `paymentIntegrationService` from `createHostedFormService`. It can be passed as argument to submit method. Please check my last commit